docs/aca.md

```diff
@@ -0,0 +1,120 @@
+   * Batch size (i.e. how many TCerts the user is expecting)
+   * ECert 
+   * List of attributes (e.g. Company:IBM, Position: Software Engineer)
+2. TCA sends under TLS a request to ACA to get those attributes:
+a. RequestAttributes() this request contains:
+     * ECert 
+     * Company: e.g., Hash(IBM)
+     * Position: e.g., Hash(Software Engineer)
+     * Signed with TCA private key
+3. ACA performs a lookup in his internal DB and replies one of the following messages***:
+     * You do not have any of those attributes (error)
+     * You have all attributes – returns an X.509 certificate with all those attributes and the ECert public key inside the certificate.
+     * You have a subset of the requested attributes. We will still return an X.509 certificate with just the subset.
+3.  TCA checks the validity period of the attributes and updates the list by eliminating those that are expired. Then it checks, for case b and c, how many (and which ones) of the attributes the user will actually receive inside each TCert. This information needs to be returned to the user so the user can decide whether the TCerts are useful to him or if he needs to perform further actions (i.e. issue a refreshAttributes command and request a new batch, throw an error or make use of the TCerts as they are).
+4.  TCA could have other criteria/rules to update the valid list of attributes
+a. This will not break any legacy stuff 
+5.  TCA creates TCerts and populates them with valid attributes
+6.  TCA encrypts the attributes with the keys derived from the Prekey tree
+7.  TCA returns the batch of TCerts (all TCerts in the batch have the same attributes and the validity period is the same for the entire batch)
+
+*** _We are executing an attributes refresh before this step, but once the refresh service is implemented then the user will have the responsibility of keeping his/her attributes updated._
+
+
+### gRPC ACA API
+
+1. FetchAttributes
+    
+    rpc FetchAttributes(ACAFetchAttrReq) returns (ACAFetchAttrResp);
+    
+    message ACAFetchAttrReq {
+        google.protobuf.Timestamp ts = 1;
+        Cert eCert = 2;                  // ECert of involved user.
+        Signature signature = 3;         // Signed using the ECA private key.
+    }
+
+    message ACAFetchAttrResp {
+            enum StatusCode {
+    	       SUCCESS = 000;
+    	       FAILURE = 100;
+	    }
+        StatusCode status = 1; 
+    }
+
+2. RequestAttributes
+
+
+    rpc RequestAttributes(ACAAttrReq) returns (ACAAttrResp);
+
+    message ACAAttrReq {
+        google.protobuf.Timestamp ts = 1;
+        Identity id = 2;
+        Cert eCert = 3;                                // ECert of involved user.
+        repeated TCertAttributeHash attributes = 4;    // Pairs attribute-key, attribute-value-hash
+        Signature signature = 5;                       // Signed using the TCA private key.
+    }
```

Comment:
Also, lines 78-82 are very different than lines 86-90 attempting to have all the '=' signs aligned.